File: src/Service/Equipaje.java
package Service;

import EmployeeTasks.InterfazID;

public class Equipaje implements InterfazID {

    private String idEquipaje;       
    private String idPasajero;     
    private double peso;          
    private String idVagonCarga;   
    private static final double MAXIMO_PESO = 80.0;    

    public Equipaje(String idEquipaje, String idPasajero, double peso, String idVagonCarga) {
        this.idEquipaje = idEquipaje;
        this.idPasajero= idPasajero;
        this.peso = validarPeso(peso); 
        this.idVagonCarga = idVagonCarga;
    }

    private static double validarPeso(double peso) { 
        if (peso > MAXIMO_PESO) {
            System.out.println("El peso excede el máximo permitido (" + MAXIMO_PESO + " kg). Ajustando al máximo permitido.");
            return MAXIMO_PESO;
        }
        return peso; 
    }

    @Override
    public String getId() {
        return idEquipaje;
    }

    public String getIdPasajero() {
        return idPasajero;
    }

    public double getPeso() {
        return peso;
    }

    public String getIdVagonCarga() {
        return idVagonCarga;
    }
}

File: Códigos/Servicio/Equipaje.cs
using ProyectoEstructuras.Códigos.TareasEmpleado;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProyectoEstructuras.Códigos.Servicio
{
    public class Equipaje : InterfazID
    { 
        private string idEquipaje;
        private string idPasajero;
        private double peso;
        private string idVagonCarga;
        private const double MAXIMO_PESO = 80.0;

        public string IdEquipaje { get => idEquipaje; set => idEquipaje = value; }
        public string IdPasajero { get => idPasajero; set => idPasajero = value; }
        public double Peso { get => peso; set => peso = ValidarPeso(value); }
        public string IdVagonCarga { get => idVagonCarga; set => idVagonCarga = value; }

        public Equipaje(string idEquipaje, string idPasajero, double peso, string idVagonCarga)
        {
            this.idEquipaje = idEquipaje;
            this.idPasajero = idPasajero;
            this.peso = ValidarPeso(peso);
            this.idVagonCarga = idVagonCarga;
        }

        private static double ValidarPeso(double peso)
        {
            if (peso > MAXIMO_PESO)
            {
                Console.WriteLine($"El peso excede el máximo permitido ({MAXIMO_PESO} kg). Ajustando al máximo permitido.");
                return MAXIMO_PESO;
            }
            return peso;
        }

        public string GetId()
        {
            return IdEquipaje;
        }
    }

}
